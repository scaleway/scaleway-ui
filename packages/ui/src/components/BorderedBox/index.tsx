import styled from '@emotion/styled'

/**
 * @deprecated BorderedBox and Containers where quite similar they have been merged into a single component called Card, use this component instead.
 */
export const BorderedBox = styled.div`
  padding: ${({ theme }) => theme.space['3']};
  border-radius: ${({ theme }) => theme.radii.default};
  border: 1px solid ${({ theme }) => theme.colors.neutral.borderWeak};
`
