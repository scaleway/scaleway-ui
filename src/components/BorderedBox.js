import { css } from '@emotion/core'
import React from 'react'
import { borderRadius, gray350 } from 'theming'
import { cx, sp } from 'utils'
import { Box } from './Box'

const style = p => css`
  padding: ${sp(1)(p)} ${sp(2)(p)};
  border-radius: ${borderRadius(p)};
  border: 1px solid ${gray350(p)};
`

export function BorderedBox(props) {
  return <Box css={cx(style)} {...props} />
}
