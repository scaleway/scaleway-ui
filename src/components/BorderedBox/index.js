import styled from '@emotion/styled'
import PropTypes from 'prop-types'
import React from 'react'
import { Box } from '../Box'
import { Link } from '../Link'

const StyledBox = styled(Box)`
  padding: ${({ theme: { space } }) => space['3']};
  border-radius: ${({ theme: { radii } }) => radii.default};
  border: 1px solid ${({ theme: { colors } }) => colors.gray350};

  ${({ variant, theme: { space } }) =>
    variant === 'vertical' &&
    `
      > * {
        margin-top: 0;
      }
      > * + * {
        margin-top: ${space['2']};
      }
    `}

  ${({ to, href, theme: { colors } }) =>
    (to || href) &&
    `
      transition: box-shadow 0.2s ease, border-color 0.2s ease;
      &:hover,
      &:focus {
        text-decoration: none;
        border-color: ${colors.primary};
        box-shadow: 2px 2px 14px 8px
          ${colors.gray200};
      }
    `}
`

const BorderedBox = ({ children, variant, to, href, as, ...props }) => (
  <StyledBox
    variant={variant}
    as={to || href ? Link : as}
    to={to}
    href={href}
    {...props}
  >
    {children}
  </StyledBox>
)

BorderedBox.propTypes = {
  variant: PropTypes.oneOf(['horizontal', 'vertical']),
  children: PropTypes.node.isRequired,
  to: PropTypes.string,
  href: PropTypes.string,
  as: PropTypes.string,
}

BorderedBox.defaultProps = {
  variant: 'horizontal',
  to: undefined,
  href: undefined,
  as: 'div',
}

export default BorderedBox
