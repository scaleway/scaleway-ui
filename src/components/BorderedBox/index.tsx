import styled from '@emotion/styled'

const BorderedBox = styled.div`
  padding: ${({ theme }) => theme.space['3']};
  border-radius: ${({ theme }) => theme.radii.default};
  border: 1px solid ${({ theme }) => theme.colorsDeprecated.gray350};
`

export default BorderedBox
