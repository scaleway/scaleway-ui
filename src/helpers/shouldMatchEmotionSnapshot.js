/* eslint-disable import/no-extraneous-dependencies */
import createCache from '@emotion/cache'
import { createSerializer } from '@emotion/jest'
import { CacheProvider, ThemeProvider } from '@emotion/react'
import { render } from '@testing-library/react'
import React from 'react'
import theme from '../theme'

expect.addSnapshotSerializer(createSerializer())

const emotionCache = createCache({
  key: 'cache',
})
emotionCache.compat = true

export default async (component, { transform } = {}) => {
  const node = render(
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </CacheProvider>,
  )
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
}
