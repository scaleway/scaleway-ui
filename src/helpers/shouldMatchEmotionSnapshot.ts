/* eslint-disable import/no-extraneous-dependencies */
import { createSerializer } from '@emotion/jest'
import { RenderOptions, render } from '@testing-library/react'
import { ReactNode } from 'react'
import defaultTheme from '../theme'
import renderWithTheme from './renderWithTheme'

// use only class hash (generated from css style content)
expect.addSnapshotSerializer(
  createSerializer({ classNameReplacer: className => className }),
)

interface Options {
  options?: RenderOptions
  theme?: typeof defaultTheme
  transform?: (node: ReturnType<typeof render>) => Promise<void> | void
}

export default async (
  component: ReactNode,
  { transform, options, theme }: Options = {},
): Promise<void> => {
  const node = renderWithTheme(component, options, theme)
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
  node.unmount()
}
