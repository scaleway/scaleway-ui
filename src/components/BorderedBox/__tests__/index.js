import React from 'react'
import BorderedBox from '..'
import shouldMatchEmotionSnapshot from '../../../helpers/shouldMatchEmotionSnapshot'

describe('BorderedBox', () => {
  test('renders correctly with default values', () => {
    shouldMatchEmotionSnapshot(<BorderedBox>Sample BorderedBox</BorderedBox>)
  })

  test('renders correctly as a to link', () => {
    shouldMatchEmotionSnapshot(
      <BorderedBox to="/">Link BorderedBox</BorderedBox>,
    )
  })

  test('renders correctly as a href link', () => {
    shouldMatchEmotionSnapshot(
      <BorderedBox href="/">Link BorderedBox</BorderedBox>,
    )
  })

  test('renders correctly as a span', () => {
    shouldMatchEmotionSnapshot(
      <BorderedBox as="span">Span BorderedBox</BorderedBox>,
    )
  })

  test('renders correctly with multiple vertical variant', () => {
    shouldMatchEmotionSnapshot(
      <>
        <BorderedBox variant="vertical">Vertical BorderedBox</BorderedBox>
        <BorderedBox variant="vertical" href="/">
          Vertical BorderedBox
        </BorderedBox>
        <BorderedBox variant="vertical" to="/">
          Vertical BorderedBox
        </BorderedBox>
      </>,
    )
  })
})
