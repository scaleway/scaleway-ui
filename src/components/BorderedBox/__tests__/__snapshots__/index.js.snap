// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderedBox renders correctly as a href link 1`] = `
<DocumentFragment>
  .emotion-1 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 200ms ease;
  transition: color 200ms ease;
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
  -webkit-transition: box-shadow 0.2s ease,border-color 0.2s ease;
  transition: box-shadow 0.2s ease,border-color 0.2s ease;
}

.emotion-1:hover,
.emotion-1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1:hover,
.emotion-1:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: #4f0599;
  box-shadow: 2px 2px 14px 8px #eeeeff;
}

<a
    class="emotion-0 emotion-1"
    href="/"
  >
    Link BorderedBox
  </a>
</DocumentFragment>
`;

exports[`BorderedBox renders correctly as a span 1`] = `
<DocumentFragment>
  .emotion-1 {
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
}

<span
    class="emotion-0 emotion-1"
    variant="horizontal"
  >
    Span BorderedBox
  </span>
</DocumentFragment>
`;

exports[`BorderedBox renders correctly as a to link 1`] = `
<DocumentFragment>
  .emotion-1 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 200ms ease;
  transition: color 200ms ease;
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
  -webkit-transition: box-shadow 0.2s ease,border-color 0.2s ease;
  transition: box-shadow 0.2s ease,border-color 0.2s ease;
}

.emotion-1:hover,
.emotion-1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1:hover,
.emotion-1:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: #4f0599;
  box-shadow: 2px 2px 14px 8px #eeeeff;
}

<a
    class="emotion-0 emotion-1"
    href="/"
    to="/"
  >
    Link BorderedBox
  </a>
</DocumentFragment>
`;

exports[`BorderedBox renders correctly with default values 1`] = `
<DocumentFragment>
  .emotion-1 {
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
}

<div
    class="emotion-0 emotion-1"
    variant="horizontal"
  >
    Sample BorderedBox
  </div>
</DocumentFragment>
`;

exports[`BorderedBox renders correctly with multiple vertical variant 1`] = `
<DocumentFragment>
  .emotion-1 {
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
}

.emotion-1>* {
  margin-top: 0;
}

.emotion-1>*+* {
  margin-top: 16px;
}

<div
    class="emotion-0 emotion-1"
    variant="vertical"
  >
    Vertical BorderedBox
  </div>
  .emotion-1 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 200ms ease;
  transition: color 200ms ease;
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
  -webkit-transition: box-shadow 0.2s ease,border-color 0.2s ease;
  transition: box-shadow 0.2s ease,border-color 0.2s ease;
}

.emotion-1:hover,
.emotion-1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1>* {
  margin-top: 0;
}

.emotion-1>*+* {
  margin-top: 16px;
}

.emotion-1:hover,
.emotion-1:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: #4f0599;
  box-shadow: 2px 2px 14px 8px #eeeeff;
}

<a
    class="emotion-0 emotion-1"
    href="/"
  >
    Vertical BorderedBox
  </a>
  .emotion-1 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 200ms ease;
  transition: color 200ms ease;
  padding: 24px;
  border-radius: 4px;
  border: 1px solid #d4dae7;
  -webkit-transition: box-shadow 0.2s ease,border-color 0.2s ease;
  transition: box-shadow 0.2s ease,border-color 0.2s ease;
}

.emotion-1:hover,
.emotion-1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1>* {
  margin-top: 0;
}

.emotion-1>*+* {
  margin-top: 16px;
}

.emotion-1:hover,
.emotion-1:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: #4f0599;
  box-shadow: 2px 2px 14px 8px #eeeeff;
}

<a
    class="emotion-0 emotion-1"
    href="/"
    to="/"
  >
    Vertical BorderedBox
  </a>
</DocumentFragment>
`;
